storage/bucket.go

```diff
@@ -801,6 +805,36 @@ func newBucket(b *raw.Bucket) (*BucketAttrs, error) {
 	}, nil
 }
 
+func newBucketFromProto(b *storagepb.Bucket) *BucketAttrs {
+	if b == nil {
+		return nil
+	}
+	return &BucketAttrs{
+		Name:                     parseBucketName(b.GetName()),
+		Location:                 b.GetLocation(),
+		MetaGeneration:           b.GetMetageneration(),
+		DefaultEventBasedHold:    b.GetDefaultEventBasedHold(),
+		StorageClass:             b.GetStorageClass(),
+		Created:                  b.GetCreateTime().AsTime(),
```

Comment:
I am wondering if some of these get methods with chained method calls need nil checks. Based on my reading of the generated code and protobuf code it sounds like maybe not in this case, but I'm wondering if we have a general guarantee/contract.

https://github.com/googleapis/go-genproto/blob/main/googleapis/storage/v2/storage.pb.go#L4186 (no nil check)
https://github.com/protocolbuffers/protobuf-go/blob/v1.27.1/types/known/timestamppb/timestamp.pb.go#L206
https://github.com/protocolbuffers/protobuf-go/blob/b92717ecb630d4a4824b372bf98c729d87311a4d/types/known/durationpb/duration.pb.go#L283 (but underlying proto code seems to treat nil as 0).

Reply:
> but I'm wondering if we have a general guarantee/contract.

I'm not sure what that means concretely, but your understanding of the code is correct - the default empty value for a field is returned in the case where receiver (proto message) is `nil`, so no need to do `nil` checks generally, because if left unset, the default empty value of the field in structure we are building would be set to that anyways.

An unset `time.Time` field is the same as a `time.Time` being created with `0`s.

Reply:
Sounds good, yeah that answers my question.